examples/stats_models/regression.jl

```diff
@@ -0,0 +1,243 @@
+
+using SemanticModels
+using Random
+using LightGraphs
+using MetaGraphs
+using DataFrames
+using GraphPlot
+using DifferentialEquations
+using Plots
+using MultivariateStats
+using Statistics
+
+mutable struct DepVar{T}
+    value::T
+end
+
+mutable struct IndepVar{T}
+    value::T
+end
+
+# to allow transformations that feature numeric elements/vectors 
+mutable struct Other{T}
+    value::T
+end
+
+
+const Var = Union{DepVar, IndepVar}
+const RegComponents = Union{DepVar, IndepVar, Other}
+
+
+mutable struct OpElementTypes{a, b}
+    vec_a::a
+    vec_b::b
+end
+
+function var_type_check(a::Var, b::Other)
+    return isa(a, DepVar) ? DepVar : IndepVar
+end
+    
+function var_type_check(a::Other, b::Var)
+    return isa(b, DepVar) ? DepVar : IndepVar
+end
+
+function var_type_check(a::Var, b::Var)   
+    return isa(a, DepVar) || isa(b, DepVar) ? DepVar : IndepVar
+
+end
+
+function var_type_check(a::Other, b::Other)   
+    return Other
+end
+
+
+binary_ops = [+,-,*,/]
+
+# example loops to get element-level truth tables 
+dv_num = 10
+iv_num = 3
+other_num = 7
+
+
+for op in binary_ops
+    for a in (DepVar(dv_num), IndepVar(iv_num), Other(other_num))
+        for b in (DepVar(dv_num), IndepVar(iv_num), Other(other_num))
+            output_check = var_type_check(a,b)
+            result_type = output_check(op(a.value, b.value))
+            print("Op: $op, a: $a, b: $b, result_type: $result_type \n")
+        end
+    end
+end
+
+function transform_col_vector(op, vec_a::Array, vec_b::Array)
```

Comment:
If you do
 
```julia
function transform(op::Function, a::Array{T,N}, b::Array{U,M}) where {T<:Var, U<:Var, N<:Integer, M<:Integer}
...
end
```

Then you can use T and U as variables and know that everything in the array `a` will have the same type (status as dep or indep) and everything in the array `b` will have the same status.

Reply:
I'm having the same problem here as w/the other cases where I try to let it be \in {DepVar, IndepVar, Other} but that those objects can take any x<:Number. trying the work around you posted earlier